### DISCUSSION TITLE
Can't install PWA on Android (insecure?)

### DISCUSSION THREAD

**Discussion by matknowles** [blakeblackshear/frigate#15655] [Ask A Question]
Hello, I don't see the option to install the PWA on Android. I'm using a Samsung A55 5G and have tried in mobile Chrome, Edge and Firefox. Looking at the requirements it looks like I might not be running in a secure context. I am using https://192.168.1.205:8971 and authenticating, but I haven't installed any certificates and do get a warning about insecure self signed certificate, so I'm assuming this is it?

Can someone help me out with how to sort this please? Thanks a lot

**Comment by NickM-27** [comment]
Yes, that is not a secure connection. You have a few options:

1. You can try going into chrome flags and enabling `trust insecure` and add `http://192.168.1.205:8971` and see if that works, some phones fail to create a PWA from an ip address:port directly (there are workarounds for this as well).
2. You can [setup a reverse proxy](https://docs.frigate.video/guides/reverse_proxy) or tunnel service to be able to access Frigate securely

**Reply by matknowles** [reply]
Thanks very much :)

**Comment by matknowles** [comment]
Can somebody check Claude AI's logic on this before I do it please as I don't want to brick my system. (for setting it up in Linux Mint 22)

`sudo apt update
sudo apt install nginx openssl`

`sudo openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /etc/ssl/private/nginx-selfsigned.key -out /etc/ssl/certs/nginx-selfsigned.crt`

Create Nginx config file /etc/nginx/sites-available/frigate

`server {
    listen 443 ssl;
    server_name frigate.local;
    ssl_certificate /etc/ssl/certs/nginx-selfsigned.crt;
    ssl_certificate_key /etc/ssl/private/nginx-selfsigned.key;

location / {
    proxy_pass http://localhost:8971;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
}
}`

`sudo ln -s /etc/nginx/sites-available/frigate /etc/nginx/sites-enabled/
sudo nginx -t
sudo systemctl restart nginx`

**Reply by NickM-27** [reply]
it might be easier to do in a docker container instead

**Comment by matknowles** [comment]
I've been tinkering with this for a while but am still confused by it. Is it possible to set this up to install the PWA for use in my LAN without purchasing/registering a domain? I tried a simple setup with Caddy and the browser still didn't allow the installation (self signed), but I'm not 100% I set it up correctly and gave up and thought I'd ask for more info before I waste any more time. I don't need or want access over the internet and if I do need to register/purchase a domain I don't think it's worth exposing my frigate to the internet just to use the PWA locally.

**Reply by NickM-27** [reply]
Owning a domain to get an SSL cert for the domain and exposing Frigate are not mutually inclusive. You can own a domain and get the cert without exposing anything over the internet.

**Reply by NickM-27** [reply]
I believe you'll either need to do that or edit the settings in your browser to trust the Frigate site despite the self signed cert

**Reply by matknowles** [reply]
ok great thanks for the info i'll stick with it then.
